server/src/lsp-server.spec.ts

```diff
@@ -155,6 +155,41 @@ describe('diagnostics', () => {
         assert.equal("Cannot find name 'missing'.", fileDiagnostics[0].message);
     }).timeout(10000);
 
+    it('supports diagnostic tags', async () => {
+        const doc = {
+            uri: uri('diagnosticsBar.ts'),
+            languageId: 'typescript',
+            version: 1,
+            text: `
+        import { join } from 'path';
+        /**
+         * @deprecated
+         */
+        function foo(): void {
+        }
+
+        foo();
+      `
+        };
+        server.didOpenTextDocument({
+            textDocument: doc
+        });
+
+        await server.requestDiagnostics();
+        await new Promise(resolve => setTimeout(resolve, 200));
+        // assert.deepEqual(JSON.stringify(diagnostics, null, 2), '');
+        const resultsForFile = diagnostics.find(d => d!.uri === doc.uri);
```

Comment:
```suggestion
        const resultsForFile = diagnostics.find(d => d?.uri === doc.uri);
```

Reply:
I think it should be fine to remove `undefined` type form the `diagnostics` array instead.